main/ruletester/ruletester.go

```diff
@@ -72,132 +105,295 @@ func main() {
 	if err != nil {
 		common.ExitWithMessage(fmt.Sprintf("Error while reading rules: %s", err.Error()))
 	}
-	_ = util.RuleBasedGraphiteConverter{Ruleset: ruleset}
+	graphiteConverter := util.RuleBasedGraphiteConverter{Ruleset: ruleset}
+
+	// metricFile, err := os.Open(*metricsFile)
+	metrics, err := ReadMetricsFile(*metricsFile)
+	if err != nil {
+		fmt.Printf("Fatal error reading metrics file %s\n", err)
+		os.Exit(1)
+	}
+
+	DoAnalysis(metrics, graphiteConverter)
+
+	fmt.Printf("Total metric count %d\n", len(metrics))
+
+	// matched := 0
+	// unmatched := 0
+	// reverse_convert_failed := 0
+
+	// fmt.Printf("Matched: %d\n", matched)
+	// fmt.Printf("Unmatched: %d\n", unmatched)
+	// fmt.Printf("Reverse convert failed: %d\n", reverse_convert_failed)
 
-	metricFile, err := os.Open(*metricsFile)
 	if err != nil {
```

Comment:
This error doesn't correspond to anything anymore (the previous call was deleted)